### PR TITLE
bsp: cvitek: fix cv18xx_aarch64 mnt init blocking

### PR DESCRIPTION
**Pull request by Z8MAN8**
The aarch64 core of duo on the master cannot enter the
console interface. It can only print the RT flag and hold it.

Analysis: The latest commit that can work is https://github.com/RT-Thread/rt-thread/commit/ae6a3287e6ab4121c07b1e0624b44d0e4477e14f ("Add
psoc62, 61 config"). This phenomenon will occur after adding
https://github.com/RT-Thread/rt-thread/commit/754c59a41185fadd543ae2eaee518ebcbe68af5b ("[Feature] DFS mount auto by kernel parameters").
The specific reason is that when aarch bsp enables the device
tree, the current u-boot will pass in bootargs, which contains
"root=/dev/mmcblk0p2 rootwait rw", which means that the
kernel is required to wait until the rootfs in /dev/mmcblk0p2
loaded successfully. However, the current aarch64 bsp default
does not implement sdmmc device mounting, causing the
 kernel file system mounting module (rootfs_mnt_init() of
components/drivers/core/mnt.c) to enter an infinite loop waiting.

Solution: At present, we do not plan to modify the startup
parameters of u-boot. The temporary solution adopted is to
create a pseudo /dev/mmcblk0p2 device during the board
initialization process, and then cancel the pseudo device
after mnt is completed. This allows the kernel boot to be
completed successfully.

Reviewed-by: Chen Wang <unicorn_wang@outlook.com>

## 拉取/合并请求描述：(PR description)

[

#### 为什么提交这份PR (why to submit this PR)
Fixed https://github.com/RT-Thread/rt-thread/issues/9243

#### 你的解决方案是什么 (what is your solution)
伪造一个 /dev/mmcblk0p2 设备, 初始化结束之后反注册这个设备.

#### 请提供验证的bsp和config (provide the config and bsp) 

- BSP: bsp/cvitek

- .config:

- action:

]

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 代码是高质量的 Code in this PR is of high quality
- [x] 已经使用[formatting](https://github.com/mysterywolf/formatting) 等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)

### CODE REVIEW

**Comment by Z8MAN8** [comment]
@unicornx

**Comment by unicornx** [comment]
> arch64 only print version logo, unable to enter console
> 
> Analysis: [754c59a](https://github.com/RT-Thread/rt-thread/commit/754c59a41185fadd543ae2eaee518ebcbe68af5b) ("[Feature] DFS mount auto by kernel parameters") cv18xx_arch64 will infinite looping in rootfs_mnt_init because of miss bootargs.
> 
> Solution: mannually add the "/dev/mmcblk0p2" device and unregiste after inited.
> 
> Reviewed-by: Chen Wang [unicorn_wang@outlook.com](mailto:unicorn_wang@outlook.com)
> 

commit 中应该是 aarch64，不是 arch64，请仔细检查一下

另外 commit 的描述不够准确。我再总结一下 #9243 的描述如下，供参考：

问题描述：目前 master 上的代码 duo 的 aarch64 大核无法启动进入控制台界面，只能打印出 RT flag 就 hold 住了。

问题分析：
bisect 看了一下，最近一个可以 work 的 commit 是 "ae6a328: Add psoc62、61 config", 加入 "754c59a4 [Feature] DFS mount auto by kernel parameters (#8989) " 后就会出现该现象。
具体原因是当 aarch bsp 启用设备树后，当前 u-boot 会传入 bootargs，其中含有 “root=/dev/mmcblk0p2 rootwait rw ”，其含义是要求内核持续等待，直到 `/dev/mmcblk0p2` 中的 rootfs 挂载成功。但目前 aarch64 bsp 默认 bsp 并没有实现 sdmmc 的设备挂载，导致内核的文件系统挂载模块（`components/drivers/core/mnt.c` 的 `rootfs_mnt_init()`） 进入死循环等待。

解决方案：目前我们暂不打算修改 u-boot 的启动参数，采用的临时解决方案是在 board 初始化过程中创建一个 pseudo 的 /dev/mmcblk0p2 设备，等 mnt 完成后再撤销该 pseudo 设备。从而让内核启动顺利完成。

File: bsp/cvitek/cv18xx_aarch64/board/board.c [hunk 1]
```diff
@@ -81,6 +81,37 @@ void rt_hw_board_init(void)
...
+{
+    if (pseudo_mmcblk)
+    {
+        return (int)rt_device_unregister(pseudo_mmcblk);
+    }
+
+    return 0;
+}
+INIT_FS_EXPORT(pseudo_mmcblk_remove);
 #endif /* RT_USING_OFW */
```
Comment (unicornx) [inline bsp/cvitek/cv18xx_aarch64/board/board.c:115]: 这个修改对  !RT_USING_OFW 应该不需要的吧，我这里只是和你确认一下。

**Review by unicornx — CHANGES_REQUESTED** [review]
见上面的 comments。

File: bsp/cvitek/cv18xx_aarch64/board/board.c [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2006-2024, RT-Thread Development Team
+ * Copyright (c) 2006-2024 RT-Thread Development Team
```
[the remark this answers is not in the document]
Comment (Z8MAN8) [reply bsp/cvitek/cv18xx_aarch64/board/board.c:2]: 这里是用formatting脚本刷了一下，该更改可以取消

File: bsp/cvitek/cv18xx_aarch64/board/board.c (hunk 1 above)
Comment (Z8MAN8) [reply bsp/cvitek/cv18xx_aarch64/board/board.c:115]: 是的

File: bsp/cvitek/cv18xx_aarch64/board/board.c [hunk 3]
```diff
@@ -81,6 +81,37 @@ void rt_hw_board_init(void)
 {
     rt_hw_common_setup();
 }
+
+/*
+ * mannually add the "/dev/mmcblk0p2" device to avoid
+ * infinite loop when rootfs_mnt_init, after init will
+ * unregiste the device.
+ * This is a temporary workaround.
```
[the remark this answers is not in the document]
Comment (Z8MAN8) [reply bsp/cvitek/cv18xx_aarch64/board/board.c]: 好的 已修改

**Review by unicornx — APPROVED** [review]

**Review by mysterywolf — APPROVED** [review]

**Pull request merged**